components/ModEditSubmissionForm.vue

```diff
@@ -692,61 +745,133 @@ const submissionHasUnsavedChanges = () => {
+                        cityJa: submissionFormFields.cityJa.value,
+                        addressLine1Ja: submissionFormFields.addressLine1Ja.value,
+                        addressLine2Ja: submissionFormFields.addressLine2Ja.value
                     }
                 },
                 healthcareProfessionalIds: [],
                 mapLatitude: parseFloat(submissionFormFields.mapLatitude.value) || 0,
                 mapLongitude: parseFloat(submissionFormFields.mapLongitude.value) || 0
-            }
+            },
+            healthcareProfessionals: [
+                {
+                    acceptedInsurance: submissionFormFields.healthcareProfessionalAcceptedInsurances.value,
+                    degrees: submissionFormFields.healthcareProfessionalDegrees.value,
+                    specialties: submissionFormFields.healthcareProfessionalSpecialties.value,
+                    spokenLanguages: submissionFormFields.healthcareProfessionalLocales.value,
+                    names: submissionFormFields.healthCareProfessionalNameArray.value,
+                    facilityIds: []
+                }
+            ]
         }
     }
+    try {
+        const result = await moderationSubmissionStore.updateSubmission(submissionInputVariables)
+        const submissionResult = result as { updateSubmission: Submission }
+        initializeSubmissionFormValues(submissionResult.updateSubmission as Submission)
+        toast.success(t('modSubmissionForm.successMessageUpdated'))
+    } catch (error) {
+        console.error('Failed to update submission:', error)
+        toast.error(t('modSubmissionForm.errorMessageUpdated'))
+        moderationSubmissionStore.setDidMutationFail(true)
+        moderationSubmissionStore.setUpdatingSubmissionFromTopBar(false)
+    }
+}
+
+async function submitCompletedForm(e: Event) {
+    // stop the form submitting before we validate
+    e.preventDefault()
+
+    //This prevents submission of an already approved submission the backend does this but as an extra visual and check
+    if (submissionFormFields.isApproved.value) {
+        toast.info(t('modSubmissionForm.infoMessageAlreadyApproved'))
+        return
+    }
+
+    if (!submissionFormFields.isUnderReview.value) {
+        toast.info(t('modSubmissionForm.infoMessageUpdateNeeded'))
+        return
+    }
+
+    const isValidFacility = validateFacilityFields()
+    const isValidHealthcareProfessional = validateHealthcareProfessionalFields()
+
+    //This shows a toast and returns if the facility fields arent valid
+    if (!isValidFacility) {
+        toast.error(t('modSubmissionForm.errorMessageFacilityInputsInvalid'))
+        return
+    }
+
+    if (!isValidHealthcareProfessional) {
+        toast.error(t('modSubmissionForm.errorMessageHealthcareInputsInvalid'))
+        return
+    }
 
-    moderationSubmissionStore.updateSubmission(submissionInputVariables)
+    try {
+        await moderationSubmissionStore.approveSubmission()
```

Comment:
we should check the server response for errors

Reply:
We talked about this